### PR TITLE
ARM Architecture Support

### PR DESCRIPTION
**Pull request by AliaksandrRyzhou**
### What change does this PR introduce?

These changes will allow us to build docker images for ARM platforms too. 

### Why was this change needed?

* This will also lower our cloud cost as ARM is more efficient and typically less expensive.
* It will allow to use Novu on ARM Platform so is really useful for many users

### Other information (Screenshots)

* We will have to remove the base image: nikolaik/python-nodejs:python3.10-nodejs16-alpine as this couldn't be compiled for arm. Because of it it's really important to make sure that it works fine after the base image changes.
* Build times might increase significantly...

Example of what we expect to see:
Screenshot from 2024-01-31 17-19-10

Screenshot from 2024-01-31 17-19-15

### CODE REVIEW

File: apps/web/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # start build stage
-FROM nikolaik/python-nodejs:python3.10-nodejs20-alpine as builder
+FROM node:20-alpine3.19 as builder
+RUN apk add g++ make py3-pip
```
Comment (Cliftonz) [inline apps/web/Dockerfile:3]: Did you test this without python?

I am unsure if we need python in this container

File: apps/webhook/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-FROM nikolaik/python-nodejs:python3.10-nodejs20-alpine as dev_base
+FROM node:20-alpine3.19 as dev_base
+RUN apk add g++ make py3-pip
```
Comment (Cliftonz) [inline apps/webhook/Dockerfile:2]: Same here

File: apps/ws/Dockerfile [hunk 3]
```diff
@@ -1,9 +1,10 @@
-FROM node:20-alpine3.16
+FROM node:20-alpine3.19
 
 ENV NX_DAEMON=false
 
 RUN npm install -g pnpm@8.9.0 --loglevel notice
 RUN npm i pm2 -g
+RUN apk --no-cache add curl g++ make py3-pip
```
Comment (Cliftonz) [inline apps/ws/Dockerfile:7]: Why was this added?

File: apps/web/Dockerfile (hunk 1 above)
Comment (AliaksandrRyzhou) [reply apps/web/Dockerfile:3]: I assumed that python was needed where basic images include python. Also Pawel said that python is used on the build and in some images the ARM architecture build hung without it. I will recheck and let you know 😊

File: apps/webhook/Dockerfile (hunk 2 above)
Comment (AliaksandrRyzhou) [reply apps/webhook/Dockerfile:2]: Same here

File: apps/ws/Dockerfile (hunk 3 above)
Comment (AliaksandrRyzhou) [reply apps/ws/Dockerfile:7]: I think it's my bad, forgot to delete after some tests, I'll remove it

File: apps/web/Dockerfile (hunk 1 above)
Comment (Cliftonz) [reply apps/web/Dockerfile:3]: @AliaksandrRyzhou Lets double check this, I know its needed for inbound-mail for sure but I have not seen anything else in the project that needs it.

@LetItRock Are you aware of any build tools that leverage python for our project?

I want to make sure we are only doing what we need to cut build times down as much as possible, but @AliaksandrRyzhou if this takes more then 2 hours to debug lets create a ticket and look at it in the future so we can get this moved through.

Comment (AliaksandrRyzhou) [reply apps/web/Dockerfile:3]: This I get when I try to build the web service without Python
image

**Review by Cliftonz — APPROVED** [review]

**Comment by AliaksandrRyzhou** [comment]
@Cliftonz I've made a quick investigation and found out that we can't build the docker images for the ARM platform without Python. It fails even in the images where Python wasn't installed initially. It looks like some npm packages include native dependencies that need to be compiled during installation. These dependencies might be platform-specific and iinstalling Python might provide a necessary build environment for these native dependencies.

**Comment by estubmo** [comment]
I wonder if something has broken this PR, because the OS/Arch tab is no longer present in the built images on ghrc

**Comment by AliaksandrRyzhou** [comment]
> I wonder if something has broken this PR, because the OS/Arch tab is no longer present in the built images on ghrc

@estubmo Please accept our apologies. We decided to change an approach of a building process in a bit, you will see multi-architecture images soon.

**Comment by estubmo** [comment]
@AliaksandrRyzhou No problem, thanks for your reply. Do you have access to some arm64 tags I could use in the meantime, preferably for the prod images?

**Comment by udleinati** [comment]
Same issue here. @AliaksandrRyzhou if you have any tag that works on arm, it would be great! thanks

**Comment by estubmo** [comment]
Any update on this @AliaksandrRyzhou?

**Comment by paulwer** [comment]
@udleinati @estubmo we have a working production on arm with an old image, i dont know it this helps someone in the meantime:
image

**Comment by paulwer** [comment]
@AliaksandrRyzhou still no updates for us?

**Comment by Cliftonz** [comment]
@paulwer @AliaksandrRyzhou and I are not working on Novu anymore.

This is something that will hopefully be fixed in the next version. @SokratisVidros would you be able to speak to this?

**Comment by SokratisVidros** [comment]
Currently, we are reshuffling the roadmap and aiming at a stable, multi-architecture, Novu V2 docker release in early September.

**Comment by paulwer** [comment]
@SokratisVidros is this considdered in the upcomming v2 release?

**Comment by SokratisVidros** [comment]
@merrcury ☝️

**Comment by paulwer** [comment]
@merrcury @SokratisVidros does it help, when I open an other issue for better clarity on this topic or are there further informations for us now?

**Comment by merrcury** [comment]
Hey, @paulwer We are working on V2 release. We will see if we will be able to solve this with that. For community Announcement, Please check discord. [Last Announcement ](https://discord.com/channels/895029566685462578/1040040454906986578/1292917556503384195). 

Otherwise, I'll separately add this to our roadmap. Stay Tuned
